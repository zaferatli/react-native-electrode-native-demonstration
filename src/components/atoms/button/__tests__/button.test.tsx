import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';

import Button, {ButtonPropsType} from '../index';
import {Linking} from 'react-native';

const getComponent = (props: ButtonPropsType) => {
  return <Button {...props} />;
};

it('initial render button', () => {
  renderer.create(getComponent({}));
});

it('Primary Button', () => {
  const props: ButtonPropsType = {
    text: '7 Day Trial and Plan',
  };
  const RenderedComponent = renderer.create(getComponent(props)).toJSON();
  expect(RenderedComponent).toMatchSnapshot();
});

it('Link Button', () => {
  const props: ButtonPropsType = {
    text: 'Privacy Policy',
    onPress: () => Linking.openURL('https://google.com'),
  };
  const RenderedComponent = renderer.create(getComponent(props)).toJSON();
  expect(RenderedComponent).toMatchSnapshot();
});

it('Icon Button', () => {
  const props: ButtonPropsType = {
    type: 'icon',
    iconType: 'BrowserIcon',
    onPress: () => Linking.openURL('https://google.com'),
  };
  const RenderedComponent = renderer.create(getComponent(props)).toJSON();
  expect(RenderedComponent).toMatchSnapshot();
});

it('Icon Button Colored', () => {
  const props: ButtonPropsType = {
    type: 'icon',
    iconType: 'BrowserIcon',
    iconFill: 'red',
    onPress: () => Linking.openURL('https://google.com'),
  };
  const RenderedComponent = renderer.create(getComponent(props)).toJSON();
  expect(RenderedComponent).toMatchSnapshot();
});
