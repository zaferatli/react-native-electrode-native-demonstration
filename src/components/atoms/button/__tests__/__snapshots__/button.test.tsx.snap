// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon Button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": undefined,
      "borderRadius": undefined,
      "color": "#FFFFFF",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": undefined,
      "fontSize": 16,
      "fontStyle": "normal",
      "fontWeight": "600",
      "height": undefined,
      "justifyContent": "center",
      "lineHeight": undefined,
      "opacity": 1,
    }
  }
>
  <browser.svg
    fill="#FFFFFF"
    iconType="BrowserIcon"
  />
</View>
`;

exports[`Icon Button Colored 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": undefined,
      "borderRadius": undefined,
      "color": "#FFFFFF",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": undefined,
      "fontSize": 16,
      "fontStyle": "normal",
      "fontWeight": "600",
      "height": undefined,
      "justifyContent": "center",
      "lineHeight": undefined,
      "opacity": 1,
    }
  }
>
  <browser.svg
    fill="red"
    iconType="BrowserIcon"
  />
</View>
`;

exports[`Link Button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#333333FF",
      "borderRadius": 9,
      "color": "#FFFFFF",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": undefined,
      "fontSize": 16,
      "fontStyle": "normal",
      "fontWeight": "600",
      "height": 48,
      "justifyContent": "center",
      "lineHeight": undefined,
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 16,
          "fontStyle": "normal",
          "fontWeight": "600",
          "lineHeight": undefined,
        },
      ]
    }
  >
    Privacy Policy
  </Text>
</View>
`;

exports[`Primary Button 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#333333FF",
      "borderRadius": 9,
      "color": "#FFFFFF",
      "display": "flex",
      "flexDirection": "row",
      "flexWrap": undefined,
      "fontSize": 16,
      "fontStyle": "normal",
      "fontWeight": "600",
      "height": 48,
      "justifyContent": "center",
      "lineHeight": undefined,
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "fontSize": 16,
          "fontStyle": "normal",
          "fontWeight": "600",
          "lineHeight": undefined,
        },
      ]
    }
  >
    7 Day Trial and Plan
  </Text>
</View>
`;
